### PR TITLE
Move quiche.BUILD sed hackery into a genrule_cmd

### PR DESCRIPTION
**Pull request by mpwarres**
*Description*:
Move quiche.BUILD sed hackery into a genrule_cmd, to allow for easier addition of further transforms, as well as comments.
*Risk Level*: Low, only affects QUICHE platform impl files/tests, which are not yet used in Envoy.
*Testing*: Existing unit tests.
*Docs Changes*: none
*Release Notes*: none
[Optional Fixes #Issue]
[Optional *Deprecated*:]

### CODE REVIEW

**Comment by mpwarres** [comment]
FYI @danzh2010 and @htuch

**Comment by danzh2010** [comment]
LGTM, thanks for figuring this out!

**Comment by mpwarres** [comment]
Tagging associated issue #2557

File: bazel/external/quiche.genrule_cmd [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  # Extract relative path (e.g. "quic/core/quic_utils.cc") from full path in
+  # src_path (e.g. "external/com_googlesource_quiche/quic/core/quic_utils.cc").
+  src_path=$${src_file#$$src_base_dir/}
+
+  # Map to output file with quiche/ base directory inserted in path.
+  out_file=$(@D)/quiche/$$src_path
+  mkdir -p $$(dirname $$out_file)
+
+  # Apply text substitutions. -E ensures consistent behavior on Linux vs. OS X.
+  sed -E -f sed_commands $$src_file > $$out_file
```
Comment (htuch) [inline bazel/external/quiche.genrule_cmd]: Should we be quoting variable derefs, in particular with paths that might have spaces, e.g. `"$$out_file"`?

Comment (mpwarres) [reply bazel/external/quiche.genrule_cmd]: Right--done.

**Pull request merged**